DataFormats/CTPPSReco/interface/TotemRPUVPattern.h

```diff
@@ -0,0 +1,70 @@
+/****************************************************************************
+*
+* This is a part of TOTEM offline software.
+* Authors:
+*   Jan Kašpar (jan.kaspar@gmail.com)
+*
+****************************************************************************/
+
+#ifndef DataFormats_CTPPSReco_TotemRPUVPattern
+#define DataFormats_CTPPSReco_TotemRPUVPattern
+
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSReco/interface/TotemRPRecHit.h"
+
+/**
+ *\brief A linear pattern in U or V projection.
+ * The intercept b is taken at the middle of a RP:
+ *     (geometry->GetRPDevice(RPId)->translation().z())
+ * The global coordinate system is used (wrt. the beam). This is the same convention
+ * as for the 1-RP track fits.
+ **/
+class TotemRPUVPattern
+{
+  public:
+    enum ProjectionType { projInvalid, projU, projV };
+
+    TotemRPUVPattern() : projection(projInvalid), a(0.), b(0.), w(0.), fittable(false)
+    {
+    }
+
+    ProjectionType getProjection() const { return projection; }
+    void setProjection(ProjectionType p_) { projection = p_; }
+
+    double getA() const { return a; }
+    void setA(double a_) { a = a_; }
+
+    double getB() const { return b; }
+    void setB(double b_) { b = b_; }
+
+    double getW() const { return w; }
+    void setW(double w_) { w = w_; }
+
+    bool getFittable() const { return fittable; }
+    void setFittable(bool f_) { fittable = f_; }
+
+    void addHit(edm::det_id_type detId, const TotemRPRecHit &hit)
+    {
+      hits.find_or_insert(detId).push_back(hit);
+    }
+
+    const edm::DetSetVector<TotemRPRecHit>& getHits() const { return hits; }
+
+    friend bool operator< (const TotemRPUVPattern &l, const TotemRPUVPattern &r);
+
+  private:
+    ProjectionType projection;              ///< projection
+    double a;                               ///< slope in rad
+    double b;                               ///< intercept in mm
+    double w;                               ///< weight
+    bool fittable;                          ///< whether this pattern is worth including in track fits
+
+    edm::DetSetVector<TotemRPRecHit> hits;  ///< hits associated with the pattern
```

Comment:
in tracking applications we keep DetId with the hit. This allows for more flexibility.
Maybe even a stronger argument is that for tracking purposes tracks and patterns rarely (if ever) have multiple hits in the same detector (on the same surface/plane)